torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -1232,15 +1253,21 @@ Tensor TensorExprKernel::convertSymbolicOutputToCorrectStrides(
       buildErrorMessage(
           "Ouput tensor has no corresponding bufs in the fuser."));
   BufPtr buf = bufs_.at(v);
-  // output is contiguous, no work to do
-  auto stride_desc = tensorOutputStrideDesc_[v->offset()];
-  if (stride_desc == torch::jit::StrideInput::TENSOR_CONT) {
+  TORCH_INTERNAL_ASSERT(buf != nullptr);
+  TORCH_INTERNAL_ASSERT(tt != nullptr);
+  TORCH_INTERNAL_ASSERT(tt->symbolic_sizes().rank() != c10::nullopt);
+
+  auto stride_desc = getSymbolicOutputStrideDesc(v);
+  auto memory_format = (stride_desc == torch::jit::StrideInput::TENSOR_CONT)
+      ? at::MemoryFormat::Contiguous
+      : at::MemoryFormat::ChannelsLast;
+  // output is contiguous with specified memory format, no work to do
+  if (buf->is_contiguous(memory_format)) {
     return Tensor(buf, nullptr);
   }
 
   TORCH_INTERNAL_ASSERT(
-      tensorOutputStrideDesc_[v->offset()] ==
-      torch::jit::StrideInput::TENSOR_CONT_CHANNELS_LAST);
+      stride_desc == torch::jit::StrideInput::TENSOR_CONT_CHANNELS_LAST);
```

Comment:
Nit: the name `StrideInput` is no longer accurate since it's now used for outputs too.

Reply:
Besides that, the `StrideInput::TENSOR_CONT` and `StrideInput::TENSOR_CONT_CHANNELS_LAST` describe the stride information at the Tensor level, while the other enum values describe the stride information at the dimension level. https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/symbolic_shape_runtime_fusion.h#L35 It might be kind of confusing. So I'm wondering if we should always describe stride per dimension and use the utility functions to query memory layout.